src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_app_extension.cc

```diff
@@ -123,7 +125,43 @@ void RCAppExtension::SaveResumptionData(
 
 void RCAppExtension::ProcessResumption(
     const smart_objects::SmartObject& saved_app,
-    resumption::Subscriber subscriber) {}
+    resumption::Subscriber subscriber) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  if (!saved_app.keyExists(
+          application_manager::strings::application_subscriptions)) {
+    LOG4CXX_DEBUG(logger_, "application_subscriptions section is not exists");
+    return;
+  }
+
+  const smart_objects::SmartObject& resumption_data =
+      saved_app[application_manager::strings::application_subscriptions];
+
+  if (!resumption_data.keyExists(message_params::kModuleData)) {
+    LOG4CXX_DEBUG(logger_, "kModuleData section does not exist");
+    return;
+  }
+
+  auto& module_data = resumption_data[message_params::kModuleData];
+
+  const auto lenght = module_data.length();
+
+  if (!lenght) {
+    LOG4CXX_DEBUG(logger_, "Subscribed modules data is absent");
+    return;
+  }
+
+  for (uint32_t index = 0; index < lenght; ++index) {
+    auto module_so = module_data[index];
+    const auto module_type = module_so[message_params::kModuleType].asString();
+    const auto module_id = module_so[message_params::kModuleId].asString();
+
+    ModuleUid module{module_type, module_id};
+    SubscribeToInteriorVehicleData(module);
```

Comment:
I do afraid that we a subscribing to vehicle data before HMI response. Will we delete subscription of error response?

Reply:
@LuxoftAKutsan will be deleted in RevertResumption